### ISSUE TITLE
About the performance on kitti car

### ISSUE THREAD

**Issue by hyssss** [xiaohaoChen/rrc_detection#16]
I follow your steps, train and test models, but the evaluation results in the kitti car are so strange.
![image](https://user-images.githubusercontent.com/18050360/27281591-00ab8292-551f-11e7-9b20-acee2ce7ed8f.png)
I don't understand why the AP on hard is so low.

**Comment by joonjeon** [comment]
Same issue here.

In fact, I have **downloaded the pretrained weights file** (in .caffemodel format) and tested RRC upon KITTI test set.

But the results are as follows:
![untitled](https://user-images.githubusercontent.com/29110031/28909713-80b8d562-7864-11e7-9566-f1ceeb383d02.png)

Results on Easy- and Moderate-level test subset seems to be reasonable, but on Hard-level test subset, mAP difference is about -7%.

Meanwhile, upon closer inspection, some result files have more than 20 instances of bounding boxes.
Related issue is as follows:
https://github.com/xiaohaoChen/rrc_detection/issues/27

If this is what's making the mAP so low on Hard-level test subset, then probably the issue on link above and the current issue can be merged.

**UPDATE(4 Aug 2017 AM 10:45 KST): looks like the number of bboxes isn't the problem. When I changed bbox result printout threshold from 0 to 0.02 this significantly reduced the number of bboxes but the mAP results were almost the same.**

Is there any explanation for this?

**Comment by joonjeon** [comment]
Well, @hyssss, at least I've got the possible solution to your problem.

If you have **trained and tested the model by yourself**, perhaps you should check each and every hyperparameters in `rrc_kitti_car.py`.
(e.g. number of total layers, learning rate, weight decay, etc.)

To solve OOM issue, it is possible for one to change the `branch_num` value from 4 to 3. But this actually DECREASES the total number of layers. Probably this is why the degradation of detection results occur.

On the other hand, in case of lower mAP despite using pre-trained weights provided by RRC developers, there must be something else I have missed (or unfortunately something else that RRC people forgot to tell).

Let's keep track of this issue. I hope RRC developers can help us soon.

**Comment by joonjeon** [comment]
Another potential cause for your issue: The LMDBs might be set incorrectly.

Training phase is supposed to cover all **7481** images, but in the current git repo snapshot, it is specified to be trained upon only **503** images. Perhaps this is the cause.

Hope this helps.

**Comment by Evence666** [comment]
@hyssss @joonjeon hello, could you teach me how to draw this result picture(PR Curve)? Is any tool can be used? Thank you!!

**Comment by hedes1992** [comment]
@Evence666  you need to use the [evaluation_object tool](http://kitti.is.tue.mpg.de/kitti/devkit_object.zip) to get the PR curve. The details are also in the zip file

**Comment by XiongweiWu** [comment]
@joonjeon I got the identical result as yours. Another potential reason is the reported result of Car hard case is predicted by ensemble model, but anyway I am waiting the author for explanation.
